### DISCUSSION TITLE
How is Microsoft supporting this project?

### DISCUSSION THREAD

**Discussion by SebastianStehle** [dotnet/orleans#7920] [General]
When you moved the project to the dotnet Organization I also had the impression that the project gets more support from Microsoft. There is a little bit more visibility now because the documentation has been moved and the release of Orleans 4 seems to be aligned with .NET 7.

But when I look to the contributors and contributions I have the impression that 90% of all work is done by @ReubenBond. You are doing a great job and your support is fast and helpful all  the time, but it is not that much for such a big company. If a few large Microsoft applications are backend by Orleans I would expect that they do not rely on a single developer only for such an important infrastructure. And it seems that not more than 1,5 developers are paid by Microsoft to work on Orleans?

The success stories from Microsoft seem to be very old as well. I only hear about Halo but I am not sure if this is still the case. I do not want to criticize the contributors, just want to understand it.

**Comment by bradygaster** [comment]
The whole Orleans community would support your perspective that @ReubenBond has been a visible and committed team member. In fact, there are a host of other folks, like @benjaminpetit, who dev on it, and with Orleans coming closer to the ASP.NET "family," the whole ASP.NET team has been getting involved, so you'll see much more contribution from the other members of the team. 

Also, there is a large series of internal apps (like Halo, which you mention, and Mesh, who have talked about their usage of Orleans on our PaaS services) - an ever-growing list, in fact, of rather happy partners and internal customers (obviously I can't speak to external customers but you can find them in our Discord, which has doubled in size recently with the renewed interest and commitments in Orleans). 

Microsoft, and specifically the .NET and ASP.NET teams, are supporting and evolving the Orleans codebase. We're shoring up official support-support this year, too, so folks can call Microsoft support and get support using it, or work with our talented field engineers to get support building apps with it. 

Hope that helps clarify our commitment to Orleans. 

One more thing, here's a great video on Orleans at Microsoft: https://www.youtube.com/watch?v=KhgYlvGLv9c

**Comment by SebastianStehle** [comment]
I really like Orleans and I have contributed to dashboard and mongo provider but for me the status quo looks like a typical Open Source project that is tied together by a few people. I am not even sure how many full time developers work on Orleans. At the moment I just do not see the increased interest from Microsoft.

**Reply by bradygaster** [reply]
On the increased interest part - it's there. We've enhanced the docs, worked with authors to get more textbooks out on the topic, and have been adding it to our agendas at conferences. There was never been more energy around Orleans than right now.

**Reply by SebastianStehle** [reply]
Does not help to decrease the bus factor

**Comment by bradygaster** [comment]
The same folks who work on ASP.NET work on Orleans. Whilst it is a light crew, it is all the crew does. And also, we're going to be offering the same level of support for Orleans that we offer for .NET over the next few months. We're in the process of training our support folks on the tech so they're more equipped to help with troubleshooting.

**Reply by ElanHasson** [reply]
Curious, what exactly is this? Paid support? The social msdn site?

**Reply by bradygaster** [reply]
Paid support. Like, you call with an issue and the reps can support your Orleans needs. Or, rather, will be. :)

**Comment by PaskeS** [comment]
Just to continue this thread around support from microsoft and Orleans v7

For those of us who were earlier adopters of Orleans e.g. 3.x and below it seems like we are bit hard done by
referencing this https://learn.microsoft.com/en-us/dotnet/orleans/whats-new-in-orleans#migration

It would be good to get some view on what this means "We plan to incrementally offer a migration path for these applications" and what sort of time frame this entails?
Is anyone at microsoft actively working on this? 
Is there an issue in place to track migration work as I was unable to find one?

**Comment by bradygaster** [comment]
There is a team actively working on Orleans at Microsoft. Specifically it is the Orleans team, which is now a sub-team of the broader .NET team. Yes, it is getting daily support and updates. No, we are not prioritizing an easy path for migration in an immediate release. We *are* working on guidance in the form of documentation (and we've already started splitting the docs into 3.x and 7.0 versions so folks can switch back and forth as they can with other Microsoft SDKs and products), and we *are* going to work on deeper migration tools and resources during the 8.0 release, but I have *no* time frames or ETA on when those will be available. 

If there are specific concerns you have around migration, let's start a separate issue, close this one, and create a list of checkboxes the community needs, so we can prioritize the efforts around what folks need to simplify migration. 

How would that sound?

**Reply by PaskeS** [reply]
That sounds great - Any visibility over what the biggest issues are in the migration and someway of tracking it would be perfect
